controllers/controllers.go

```diff
@@ -66,3 +79,39 @@ func mergeMap(target map[string]string, desired map[string]string) bool {
 	}
 	return changed
 }
+
+// Requeue provides a wrapper around different results from a subreconciler.
+type Requeue struct {
+	// Delay provides an optional delay before requeueing reconciliattion.
+	Delay time.Duration
+
+	// Error provides an error that we encountered that forced a requeue.
+	Error error
+
+	// Message provides a log message that explains the reason for the requeue.
+	Message string
+}
+
+// processRequeue interprets a requeue resulit from a subreconciler.
+func processRequeue(requeue *Requeue, subReconciler interface{}, object runtime.Object, recorder record.EventRecorder, logger logr.Logger) (ctrl.Result, error) {
```

Comment:
Before we aborted early when the generation was changed win the current implementation we ignore that. I guess that's intended?

Reply:
I think checking the generation was a legacy of when the operator would post updates to the spec of the cluster itself, and that's something we want to prevent.